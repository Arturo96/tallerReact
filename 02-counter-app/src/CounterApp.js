import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({value = 10}) => {

    const [counter, setCounter] = useState(value);

    const incrementar = (e) => setCounter(counter + 1);
    const resetear = (e) => setCounter(value);
    const decrementar = (e) => setCounter(counter - 1);
    
    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <div>
                <button onClick={incrementar} >+1</button>
                <button onClick={resetear} >Reset</button>
                <button onClick={decrementar} >-1</button>
            </div>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;
