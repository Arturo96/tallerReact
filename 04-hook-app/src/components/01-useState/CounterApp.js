import React, { useState } from 'react'

export const CounterApp = () => {

    const [state, setState] = useState({
        counter1: 10,
        counter2: 20,
        counter3: 30,
        counter4: 40
    });

    const {counter1, counter2} = state;

    return (

        <div className="container mt-3">
         <h2>Counter1 {counter1}</h2>
         <h2>Counter2 {counter2}</h2>
         <hr /> 

         <button onClick={() => setState({
             ...state,
             counter1: counter1 + 1
         })} className="btn btn-primary mr-3">+1</button>
        </div>
    )
}
