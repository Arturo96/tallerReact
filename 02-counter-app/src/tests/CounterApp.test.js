import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import CounterApp from '../CounterApp';
import React from 'react';

describe('Pruebas del componente CounterApp', () => {

    let wrapper = shallow(<CounterApp />);

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    })

    test('Debe de mostrar <CounterApp /> con sus valores default', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar el contador por defecto 100', () => {
        const wrapper = shallow(<CounterApp value={ 100 } />);

        const counter = parseInt(wrapper.find('h2').text());

        expect(counter).toBe(100);
    })
    
    test('Debe de incrementar con el botón +1', () => {
        wrapper.find('button').at(0).simulate('click');

        const counter = parseInt(wrapper.find('h2').text());

        expect(counter).toBe(11);



    })

    test('Debe de decrementar con el botón -1', () => {

        wrapper.find('button').at(2).simulate('click');

        const counter = parseInt(wrapper.find('h2').text());

        expect(counter).toBe(9);

    })
    
    test('Debe de resetear el valor con el botón Reset', () => {

        wrapper = shallow(<CounterApp value={ 105 } />);

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const counter = parseInt(wrapper.find('h2').text());

        expect(counter).toBe(105);

    })
    
})