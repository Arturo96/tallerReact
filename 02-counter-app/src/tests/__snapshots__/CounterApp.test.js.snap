// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente CounterApp Debe de mostrar <CounterApp /> con sus valores default 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <div>
    <button
      onClick={[Function]}
    >
      +1
    </button>
    <button
      onClick={[Function]}
    >
      Reset
    </button>
    <button
      onClick={[Function]}
    >
      -1
    </button>
  </div>
</Fragment>
`;
